addons/stock_dropshipping/tests/test_dropship.py

```diff
@@ -121,3 +121,63 @@ def test_00_dropship(self):
             ('location_dest_id', '=', self.env.ref('stock.stock_location_customers').id),
             ('product_id', '=', drop_shop_product.id)])
         self.assertEquals(len(move_line.ids), 1, 'There should be exactly one move line')
+
+    def test_01_dropship_with_different_uom(self):
+        
+        # Create a vendor
+        supplier_dropship = self.env['res.partner'].create({'name': 'Vendor of Dropshipping test'})
+        
+        # Create a product
+        test_product = self.env['product.template'].create({"name": "Product"})
+        
+        # Create a component with UoM Liters with dropshipping route
+        dropshipping_route = self.env.ref('stock_dropshipping.route_drop_shipping')
+        component_liter = self.env['product.template'].create({
+            "name": "component liter",
+            "route_ids": [(6, 0, [dropshipping_route.id])],
+            "seller_ids": [(0, 0, {
+                'delay': 1,
+                'name': supplier_dropship.id,
+                'min_qty': 1.0
+            })],
+            "uom_id": 11,
+            "uom_po_id": 11,
+        })
+        
+        # Create an other component with UoM Units without dropshipping route
+        component_unit = self.env['product.template'].create({
+            "name": "component liter",
+            "uom_id": 1,
+            "uom_po_id": 1,
+        })
+        
+        # Create a BoM for the product with the two components
+        self.env['mrp.bom'].create({
+            "product_tmpl_id": test_product.id,
+            "product_id": False,
+            "product_qty": 1,
+            "type": "phantom",
+            "bom_line_ids": [
+                [0, 0, {"product_id": component_liter.product_variant_id.id, "product_qty": 1}],
+                [0, 0, {"product_id": component_unit.product_variant_id.id, "product_qty": 1}],
+            ]
+        })
+        
+        # Create a sales order with a line of 1 test_product
+        so_form = Form(self.env['sale.order'])
+        so_form.partner_id = self.env.ref('base.res_partner_2')
+        so_form.payment_term_id = self.env.ref('account.account_payment_term_end_following_month')
+        with mute_logger('odoo.tests.common.onchange'):
```

Comment:
instead you can add quantity at the beginning of the test. It would be cleaner

Reply:
You mean move the so_form at the beginning ?

Reply:
self.env['stock.quant']._update_available_quantity(...)

Reply:
Done, thanks